src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -297,6 +298,13 @@
           (if (nil? body-string)
             (http/error-response (str "Empty body for application/json submission"))
             (app (assoc req :body-string body-string))))
+        "multipart/form-data"
+        (app (let [mp-request (mp/multipart-params-request req)
```

Comment:
It appears that this isn't actually streaming. Rather, it'll put the data into a file, which we don't really want to do. See https://github.com/ring-clojure/ring/blob/1.4.0/ring-core/src/ring/middleware/multipart_params.clj#L79 and https://github.com/ring-clojure/ring/blob/1.4.0/ring-core/src/ring/middleware/multipart_params/temp_file.clj 

The out-of-the-box alternative loads into a byte array, which is at least better, but still not great.